### PR TITLE
Create Monitor - with JSON schema

### PR DESCRIPTION
**Pull request by FarajiA**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-201

 ## Description:
**Add CPU Monitor functionality**

- CPU Monitor
  1. Dropdown option for selecting CPU
  2. Add new create monitor service request

- Schema
  1. Add new Service for Schema
  2. Add new mock for schema
  3. Add new Interface for schema
  4. Add new interfaces for Monitor based on Schema
      - https://bcherny.github.io/json-schema-to-typescript-browser (used for generating the interfaces for creating a new monitor)

  5. [AJV Schema validation](https://ajv.js.org) added to project
      - Validates JSON body before the data is posted to the API    

- Labels
  1. Add new Service for Labels (Resources & Monitors)
  2. Add capability for components to display Datalist with Label keys & values

 ## Testing:
`npm run test`

 ## Screenshots:
createCPUMonitor

### CODE REVIEW

File: src/app/_models/monitors.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@
 interface Label {
     [key: string] : any
 }
-export interface Monitor {
```
Comment (pras7430) [inline src/app/_models/monitors.ts:5]: We can use Pascal casing for interface as it is  a best practice.

File: localenv/package.json [hunk 2]
```diff
@@ -35,6 +34,7 @@
     ]
   },
   "dependencies": {
+    "axios": "^0.19.2",
```
Comment (pras7430) [inline localenv/package.json:37]: I think we should use fixed versions

File: src/app/_services/monitors/schema.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should set & get schema', () => {
+    service.schema = new monitorsMock().schema;
+    expect(service.schema.description).toEqual("Salus Monitor definition");
+  });
+
+  it('should load schema file', () => {
```
Comment (pras7430) [inline src/app/_services/monitors/schema.service.spec.ts:37]: we can exclude or remove it.

File: src/app/_services/labels/label.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  });
+
+  it('should set and get Monitor labels', () => {
+    service.labels = new LabelMock().monitorLabels;
+    expect(service.labels.agent_discovered_os[0]).toEqual("linux");
+  });
+
+  it('should set and get Resource labels', () => {
+    service.labels = new LabelMock().resourceLabels;
+    expect(service.labels.agent_discovered_hostname[0]);
```
Comment (pras7430) [inline src/app/_services/labels/label.service.spec.ts:30]: should we validate the expected value or it is okay?

File: src/app/_models/labels.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+export interface ILabelResources extends ILabelMonitors{
```
Comment (pras7430) [inline src/app/_models/labels.ts:1]: can we use pascal name all the properties and can make a mapper where we can assign such our properties with service data.... 
like
labelMonitor.agentDiscoveredOs= service.data.agent_discovered_os;

File: src/app/_features/monitors/mon.utils.spec.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            cpu: {
+                percpu: true,
+                totalcpu: false,
+                collectCpuTime: false,
+                reportActive: true
+            }
+        }
+        let cpuMonitor = FormatMonitorUtil('cpu', monitor);
+        expect(cpuMonitor.details.type).toEqual('local');
+        expect(cpuMonitor.details.plugin.type).toEqual('cpu');
```
Comment (pras7430) [inline src/app/_features/monitors/mon.utils.spec.ts:52]: i think it is duplicate and hard code value check

Comment (FarajiA) [reply src/app/_features/monitors/mon.utils.spec.ts:52]: Yup, this was updated.

File: src/app/_services/labels/label.service.spec.ts (hunk 4 above)
Comment (FarajiA) [reply src/app/_services/labels/label.service.spec.ts:30]: Yup, updated.

File: src/app/_models/labels.ts (hunk 5 above)
Comment (FarajiA) [reply src/app/_models/labels.ts:1]: I'm hoping you could refactor and show what you mean.

File: src/app/_services/monitors/schema.service.spec.ts (hunk 3 above)
Comment (FarajiA) [reply src/app/_services/monitors/schema.service.spec.ts:37]: Yes, good catch this test was incomplete.

File: localenv/package.json (hunk 2 above)
Comment (FarajiA) [reply localenv/package.json:37]: Done.

File: src/app/_models/monitors.ts (hunk 1 above)
Comment (FarajiA) [reply src/app/_models/monitors.ts:5]: Pascal yes, removed the "I" from the interfaces as well.

**Pull request merged**